### ISSUE TITLE
Unable to get resourcegroup, it is created but cannot be seen with kubectl get

### ISSUE THREAD

**Issue by shalinin00** [Azure/azure-service-operator#3329]
**Version of Azure Service Operator**

**Describe the bug**
Fyi - I've created managed identity and also federated identity.
I am trying to create resource group by helm install ASOv2 for Managed Identity. 
Trying to create a resource group after following this wiki: https://github.com/Azure/azure-service-operator/blob/main/v2/README.md 
I have exported all the variables and installed and then all controller pods are coming up fine. 
When trying to create a resource group, getting the following error 

kubectl get resourcegroup 
No resources found in default namespace.

**To Reproduce**
Steps to reproduce the behavior: Follow the instructions in the wiki https://github.com/Azure/azure-service-operator/blob/main/v2/README.md 
Trying to create Resource group step.
When I do  ```
kubectl get resourcegroup 
No resources found in default namespace.
```
kubectl logs -f azureserviceoperator-controller-manager9806-ljllp -n azureserviceoperator-system
```
2023/09/23 03:51:33 http: TLS handshake error from 173.74.2.34.47:9876353: EOF
2023/09/23 03:51:33 http: TLS handshake error from 173.74.2.34.47:9876353: EOF
2023/09/23 03:51:33 http: TLS handshake error from 173.74.2.34.47:9876353: EOF
2023/09/23 03:51:33 http: TLS handshake error from 173.74.2.34.47:9876353: EOF
I0923 03:51:34.384007       1 common.go:58] controllers/ResourceGroupController "msg"="Reconcile invoked" "annotations"={"serviceoperator.azure.com/operator-namespace":"azureserviceoperator-system"} "conditions"="[]" "creationTimestamp"="2023-09-23T03:51:33Z" "deletionTimestamp"=null "finalizers"=null "generation"=1 "kind"={"kind":"ResourceGroup","apiVersion":"resources.azure.com/v1api20200601storage"} "name"="aso-rg-sampletest" "namespace"="default"

```

**Expected behavior**

Resource group should be created successfully through ASOv2 with managed identity.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

**Comment by super-harsh** [comment]
Hi @shalinin00, could you please provide us with a bit more information on `helm` command you used to install ASO? 

Also, For more information on ASO authentication modes, I'd suggest you refer to ASO authentication docs [here](https://azure.github.io/azure-service-operator/guide/authentication)

**Comment by shalinin00** [comment]
Helm command - 
```
helm upgrade --install aso2 aso2/azure-service-operator \
     --create-namespace \
     --namespace=azureserviceoperator-system \
     --set azureSubscriptionID=$AZURE_SUBSCRIPTION_ID \
     --set azureTenantID=$AZURE_TENANT_ID \
     --set azureClientID=$AZURE_CLIENT_ID \
     --set useWorkloadIdentityAuth=true \
     --set crdPattern='resources.azure.com/*;containerservice.azure.com/*;keyvault.azure.com/*;managedidentity.azure.com/*;eventhub.azure.com/*;dbformysql.azure.com/*'
```

**Comment by shalinin00** [comment]
> Hi @shalinin00, could you please provide us with a bit more information on `helm` command you used to install ASO?
> 
> Also, For more information on ASO authentication modes, I'd suggest you refer to ASO authentication docs [here](https://azure.github.io/azure-service-operator/guide/authentication)

Helm command -

helm upgrade --install aso2 aso2/azure-service-operator \
     --create-namespace \
     --namespace=azureserviceoperator-system \
     --set azureSubscriptionID=$AZURE_SUBSCRIPTION_ID \
     --set azureTenantID=$AZURE_TENANT_ID \
     --set azureClientID=$AZURE_CLIENT_ID \
     --set useWorkloadIdentityAuth=true \
     --set crdPattern='resources.azure.com/*;containerservice.azure.com/*;keyvault.azure.com/*;managedidentity.azure.com/*;eventhub.azure.com/*;dbformysql.azure.com/*'

**Comment by shalinin00** [comment]
@super-harsh / Team, 
Any update on this? Im stuck in creating resource group.

**Comment by matthchr** [comment]
is that the full ASO logs? Can you share a more complete set of logs from the ASO pod? The TLS handshake error may be nothing of concern and the actual problem is somewhere else in the logs.

What version of Kubernetes are you running on? IIRC this TLS error may be due to a Go bug that impacted 1.23-1.24: https://github.com/kubernetes/kubernetes/issues/109022

**Comment by shalinin00** [comment]
Thankyou so much for the explanation. Appreciate your response. 

<img width="1103" alt="asov2-rg-unabletoget" src="https://github.com/Azure/azure-service-operator/assets/139930681/16e57c70-6c79-405f-a89c-85dc46753398">

I'm able to create resourcegroup from aso v2 controller and see it in Azure portal. But unable to GET resourcegroup. 

Full log for aso v2 controller - 
```
I0925 22:01:15.987496 1 common.go:58] controllers/ResourceGroupController
"msg"="Reconcile invoked" "annotations"={} "conditions"="[]" "creationTimestamp"="2023-09-
25T22:01:15Z" "deletionTimestamp"=null "finalizers"=null "generation"=1 "kind"=
{"kind":"ResourceGroup","apiVersion":"resources.azure.com/v1api20200601storage"}
"name"="aso-rg-sep-test" "namespace"="default" "owner"=null "ownerReferences"=null
"resourceVersion"="714303522" "uid"="262997262-9282763-928262-9272762"
2023/09/25 22:01:15 http: TLS handshake error from 172.87.9.89:879: EOF
2023/09/25 22:01:15 http: TLS handshake error from 172.87.9.89:879: EOF
I0925 22:01:17.014755 1 common.go:58] controllers/ResourceGroupController
"msg"="Reconcile invoked" "annotations"={"serviceoperator.azure.com/operatornamespace":"
azureserviceoperator-system"} "conditions"="[]" "creationTimestamp"="2023-09-
25T22:01:15Z" "deletionTimestamp"=null "finalizers"=null "generation"=1 "kind"=
{"kind":"ResourceGroup","apiVersion":"resources.azure.com/v1api20200601storage"}
"name"="aso-rg-sep-test" "namespace"="default" "owner"=null "ownerReferences"=null
"resourceVersion"="714303523" "uid"="262997262-9282763-928262-9272762"
I0925 22:01:17.014785 1 common.go:125] controllers/ResourceGroupController
"msg"="applying ownership" "name"="aso-rg-sep-test" "namespace"="default"
I0925 22:01:17.014895 1 generic_reconciler.go:186] controllers/ResourceGroupController
"msg"="adding finalizer" "name"="aso-rg-sep-test" "namespace"="default"
I0925 22:01:17.015024 1 recorder.go:104] events "msg"="Using credential from
\"azureserviceoperator-system/aso-controller-settings\"" "object"=
{"kind":"ResourceGroup","namespace":"default","name":"aso-rg-sep-test","uid":"262997262-
9282763-928262-
9272762","apiVersion":"resources.azure.com/v1api20200601storage","resourceVersion":"7143035
23"} "reason"="CredentialFrom" "type"="Normal"
I0925 22:01:17.060818 1 azure_generic_arm_reconciler_instance.go:71]
controllers/ResourceGroupController "msg"="Determined CreateOrUpdate action"
"action"="BeginCreateOrUpdate" "azureName"="aso-rg-sep-test" "name"="aso-rg-sep-test"
"namespace"="default"
I0925 22:01:17.060893 1 azure_generic_arm_reconciler_instance.go:308]
controllers/ResourceGroupController "msg"="About to send resource to Azure"
"azureName"="aso-rg-sep-test" "name"="aso-rg-sep-test" "namespace"="default"
I0925 22:01:17.060933 1 recorder.go:104] events "msg"="Using credential from
\"azureserviceoperator-system/aso-controller-settings\"" "object"=
{"kind":"ResourceGroup","namespace":"default","name":"aso-rg-sep-test","uid":"262997262-
9282763-928262-
9272762","apiVersion":"resources.azure.com/v1api20200601storage","resourceVersion":"7143035
35"} "reason"="CredentialFrom" "type"="Normal"
I0925 22:01:17.316585 1 azure_generic_arm_reconciler_instance.go:317]
controllers/ResourceGroupController "msg"="Successfully sent resource to Azure"
"azureName"="aso-rg-sep-test" "id"="/subscriptions//resourceGroups/aso-rg-sep-test"
"name"="aso-rg-sep-test" "namespace"="default"
I0925 22:01:17.316635 1 azure_generic_arm_reconciler_instance.go:414]
controllers/ResourceGroupController "msg"="Resource successfully created/updated"
"azureName"="aso-rg-sep-test" "name"="aso-rg-sep-test" "namespace"="default"
"resourceID"="/subscriptions/86478638392011/resourceGroups/aso-rg-sep-test"
I0925 22:01:17.316795 1 recorder.go:104] events "msg"="Successfully sent resource to Azure
with ID \"/subscriptions/28276393-83738302-89762572926/resourceGroups/aso-rg-septest\""
"object"={"kind":"ResourceGroup","namespace":"default","name":"aso-rg-septest","
uid":"262997262-9282763-928262-
9272762","apiVersion":"resources.azure.com/v1api20200601storage","resourceVersion":"7143035
35"} "reason"="BeginCreateOrUpdate" "type"="Normal"
I0925 22:01:17.374926 1 generic_reconciler.go:136] controllers/ResourceGroupController
"msg"="Done with reconcile" "name"="aso-rg-sep-test" "namespace"="default" "result"=
{"Requeue":false,"RequeueAfter":2879598339759} 
```

**Comment by matthchr** [comment]
Do you have ASOv1 installed as well? Or some other operator which installs a CRD called `resourcegroup`?

Can you try using the fully specified CRD name instead of the shorthand? `kubectl get resourcegroup.resources.azure.com` should show it. It has to exist because ASO reconciled it.

You can also examine `kubectl get crds` to see what 2 CRDs you have that are named `resourcegroup`. I _believe_ that Kubernetes prefers the one that was installed earlier.

**Comment by shalinin00** [comment]
@matthchr Yes, i've ASOv1 installed. Now, I can GET all the created resources. Thank you so much!!
Also, i can see 2 CRD installed. 

I hope this could be documented in the ASO v2 which will be helpful. 

[A little of context] We have v1, and installing v2 as we need to migrate from v1 to v2. 
I'm going to face the same issue for flexible server CRD as im planning to migrate from mysql to flexible server?

**Comment by matthchr** [comment]
You'll only face this issue where the actual CRD shortnames overlap. MySQL and MySQLFlexibleServer short names don't overlap, so you won't have this issue.

The only resource I am aware that has this issue in ASOv1 vs ASOv2 is ResourceGroup. It's not an "issue" so much as a confusing interaction. It's actually working as intended.

I think your problem is solved, so I've closed this issue and filed #3342 tracking some documentation improvements that should call this out more explicitly.
